bot/exts/utilities/color.py

```diff
@@ -0,0 +1,224 @@
+        """Command to create an embed from an HSL input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, lightness)):
+            raise BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Lightness can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, lightness}`."
+            )
+        hsl_tuple = ImageColor.getrgb(f"hsl({hue}, {saturation}%, {lightness}%)")
+        await self.send_colour_response(ctx, hsl_tuple)
+
+    @colour.command()
+    async def cmyk(self, ctx: commands.Context, cyan: int, magenta: int, yellow: int, key: int) -> None:
+        """Command to create an embed from a CMYK input."""
+        if any(c not in range(0, 101) for c in (cyan or magenta or yellow or key)):
+            raise BadArgument(
+                message=f"CMYK values can only be from 0 to 100. User input was: `{cyan, magenta, yellow, key}`."
+            )
+        r = round(255 * (1 - (cyan / 100)) * (1 - (key / 100)))
+        g = round(255 * (1 - (magenta / 100)) * (1 - (key / 100)))
+        b = round(255 * (1 - (yellow / 100)) * (1 - (key / 100)))
+        await self.send_colour_response(ctx, (r, g, b))
+
+    @colour.command()
+    async def hex(self, ctx: commands.Context, hex_code: str) -> None:
+        """Command to create an embed from a HEX input."""
+        if "#" not in hex_code:
+            hex_code = f"#{hex_code}"
+        hex_tuple = ImageColor.getrgb(hex_code)
+        await self.send_colour_response(ctx, hex_tuple)
+
+    @colour.command()
+    async def name(self, ctx: commands.Context, user_colour_name: str) -> None:
+        """Command to create an embed from a name input."""
+        hex_colour = self.match_colour_name(user_colour_name)
+        hex_tuple = ImageColor.getrgb(hex_colour)
+        await self.send_colour_response(ctx, hex_tuple)
+
+    @colour.command()
+    async def random(self, ctx: commands.Context) -> None:
+        """Command to create an embed from a randomly chosen colour from the reference file."""
+        hex_colour = random.choice(list(self.colour_mapping.values()))
+        hex_tuple = ImageColor.getrgb(f"#{hex_colour}")
+        await self.send_colour_response(ctx, hex_tuple)
+
+    def get_colour_conversions(self, rgb: tuple[int, int, int]) -> dict[str, str]:
+        """Create a dictionary mapping of colour types and their values."""
+        return {
+            "RGB": rgb,
+            "HSV": self._rgb_to_hsv(rgb),
+            "HSL": self._rgb_to_hsl(rgb),
+            "CMYK": self._rgb_to_cmyk(rgb),
+            "Hex": self._rgb_to_hex(rgb),
+            "Name": self._rgb_to_name(rgb)
+        }
+
+    @staticmethod
+    def _rgb_to_hsv(rgb: tuple[int, int, int]) -> tuple[int, int, int]:
+        """Convert RGB values to HSV values."""
+        rgb_list = [val / 255.0 for val in rgb]
+        h, v, s = colorsys.rgb_to_hsv(*rgb_list)
```

Comment:
The order here is wrong
```suggestion
        h, s, v = colorsys.rgb_to_hsv(*rgb_list)
```

Reply:
The values needed to be swapped, fixed with 8808914.